### PR TITLE
#156 eslinter configuration + fix and reformat existing code

### PR DESCRIPTION
**Pull request by lexich**
1. Create eslint-configuration
2. run `yarn lint:fix`
3. manual fix non-automaticaly fixing problems
4. run `yarn lint:fix` - now it's ok
5. add `eslint --cache --fix --max-warnings=0` as step for 'lint-staged'
6. add github action

https://github.com/metaplex-foundation/metaplex/issues/156

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Lint
+
+on: [push]
```
Comment (sponomarev) [inline .github/workflows/lint.yml:3]: I'm not sure that running this on every push is useful. I would suggest running linters against PRs and master branch pushes. 

```suggestion
on:
  pull_request:
  push:
    branches: [master]
```

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+jobs:
+  eslint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Install modules
+        run: yarn
+        working-directory: ./js
```
Comment (sponomarev) [inline .github/workflows/lint.yml:12]: There's a [convenient way](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#defaultsrun) to apply it globally to the whole workflow.

```
defaults:
  run:
    working-directory: js
```

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+on: [push]
+
+jobs:
+  eslint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - name: Install modules
+        run: yarn
```
Comment (sponomarev) [inline .github/workflows/lint.yml:11]: Are you 100% sure that there's no need to install Node with Yarn first?  Besides, we can [cache dependencies](https://github.com/actions/setup-node#caching-packages-dependencies)  between invocations to reduce job time.

File: js/packages/common/src/contracts/token.ts [hunk 4]
```diff
@@ -25,9 +20,9 @@ export const mintNFT = async (
   const TOKEN_PROGRAM_ID = new PublicKey(
     'TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA',
   );
-  const SPL_ASSOCIATED_TOKEN_ACCOUNT_PROGRAM_ID = new PublicKey(
-    'ATokenGPvbdGVxr1b2hvZbsiqW5xWH25efTNsLJA8knL',
-  );
+  //const SPL_ASSOCIATED_TOKEN_ACCOUNT_PROGRAM_ID = new PublicKey(
```
Comment (sponomarev) [inline js/packages/common/src/contracts/token.ts:23]: Shouldn't it be removed completelyy?

File: js/packages/common/src/models/account.ts [hunk 5]
```diff
@@ -62,7 +62,7 @@ export function approve(
   const tokenProgram = TOKEN_PROGRAM_ID;
 
   const transferAuthority = existingTransferAuthority || Keypair.generate();
-  const delegateKey = delegate ?? transferAuthority.publicKey;
+  //const delegateKey = delegate ?? transferAuthority.publicKey;
```
Comment (sponomarev) [inline js/packages/common/src/models/account.ts:65]: Ditto

File: js/packages/web/src/contexts/meta/loadAccounts.ts [hunk 6]
```diff
@@ -34,14 +34,14 @@ async function getProgramAccounts(
 ): Promise<Array<AccountAndPubkey>> {
   const extra: any = {};
   let commitment;
-  let encoding;
+  //let encoding;
```
Comment (sponomarev) [inline js/packages/web/src/contexts/meta/loadAccounts.ts:37]: Ditto

File: js/packages/web/src/contexts/meta/loadAccounts.ts [hunk 7]
```diff
@@ -34,14 +34,14 @@ async function getProgramAccounts(
...
 
   if (configOrCommitment) {
     if (typeof configOrCommitment === 'string') {
       commitment = configOrCommitment;
     } else {
       commitment = configOrCommitment.commitment;
-      encoding = configOrCommitment.encoding;
+      //encoding = configOrCommitment.encoding;
```
Comment (sponomarev) [inline js/packages/web/src/contexts/meta/loadAccounts.ts:44]: Ditto

File: js/packages/common/src/contracts/token.ts (hunk 4 above)
Comment (lexich) [reply js/packages/common/src/contracts/token.ts:23]: Not sure

File: .github/workflows/lint.yml (hunk 3 above)
Comment (lexich) [reply .github/workflows/lint.yml]: Good point

File: .github/workflows/lint.yml (hunk 2 above)
Comment (lexich) [reply .github/workflows/lint.yml]: Ok

Comment (sponomarev) [reply .github/workflows/lint.yml]: Sorry, don't see any changes yet

**Comment by lexich** [comment]
@bartosz-lipinski Can you see this PR?

**Pull request merged**